src/main/java/uk/nhs/hee/tis/trainee/forms/service/impl/FormRPartAServiceImpl.java

```diff
@@ -77,11 +77,10 @@ public FormRPartADto save(FormRPartADto formRPartADto) {
     if (alwaysStoreFiles || formRPartA.getLifecycleState() == LifecycleState.SUBMITTED
         || formRPartA.getLifecycleState() == LifecycleState.UNSUBMITTED) {
       cloudObjectRepository.save(formRPartA);
-      //Save in mongo for backward compatibility
-      repository.save(formRPartA);
-    } else {
-      formRPartA = repository.save(formRPartA);
     }
+
+    // Forms stored in cloud are still stored to Mongo for backwards compatibility.
+    formRPartA = repository.save(formRPartA);
```

Comment:
There is a subtle difference in returning the object saved in the `repository` rather than the parameter.  I think it's ok as I'm not aware of any further code that relies on the object identity.

I do prefer this simplification.

Reply:
I changed the behaviour based on the tests, they were written to verify that the ID is set but were written in such a way that the ID was being set by the test code rather than by the implementation.